app/User.php

```diff
@@ -289,4 +315,41 @@ public function scopeInactive($query)
             $q->paid()->current();
         });
     }
+
+    /**
+     * Scope a query to automatically include only access active members
+     * Active: Has paid dues for a currently ongoing term
+     *         or, has a non-zero payment for an active DuesPackage.
+     * @param \Illuminate\Database\Eloquent\Builder $query
+     * @param mixed $type
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopeAccessActive($query)
+    {
+        $now = new \DateTime();
+
+        return $query->whereHas('dues', function ($q) {
+            $q->paid()->accessCurrent();
+        })->orWhere('access_override_until', '>', date('Y-m-d'));
```

Comment:
I don't know for sure if these date semantics are correct. Eventually, a job will be fired a few seconds after `access_override_until` and at that point it should see that the user's access flag is `false`.

It's okay to expire access slightly before intended (as the job won't fire until that time anyway) but it's not okay for the job to think a user has access when they don't (as you don't know that another job will fire later.)